### ISSUE TITLE
Directory structure rationalization 

### ISSUE THREAD

**Issue by septract** [rems-project/cn-tutorial#43]
We should reorganize the directory structure to be more reasonable. Currently it looks like this: 
```
cn-tutorial/
├─ check.sh
├─ Makefile
├─ src/
│  ├─ tutorial.adoc
│  ├─ examples/
│  │  ├─ 3.c
│  │  ├─ ...
│  ├─ example-archive/
│  │  ├─ check-all.sh
│  │  ├─ check.sh
│  │  ├─ simple-examples/
│  │  │  ├─ working/
│  │  │  ├─ broken/
│  │  ├─ .../
```

Some anomalies: 
* The `example-archive` folder is buried, and we might want to lift it to the top level 
* Examples are categorized by name in `src/examples` but by folder in `src/example-archive/`

@bcpierce00

**Comment by thatplguy** [comment]
Perhaps we can move `example-archive` into the CN repo, once it's split from the Cerberus repo?  As remarked in the past, the example archive is less about teaching CN to new users and more for characterizing CN capabilities for us (tool developers) and for regression testing.

I hear from @dc-mak that the CN/Cerberus repo split is underway and will hopefully be complete in the not too distant future.

**Comment by bcpierce00** [comment]
That would make sense.  It might lead to a bit of duplication, though,
since I guess we'll want to have many of the same (larger) examples in both
places...

**Comment by thatplguy** [comment]
I think some duplication is OK?  I don't think it's a problem if the tutorial examples change independently of the example archive (e.g. improving readability, adding comments, etc.), and any structural changes we need to make, like syntax changes, will be more painful due to the number of examples in the archive rather than the relatively small bit of duplication.

**Comment by bcpierce00** [comment]
Agreed.  The advantages of splitting are more than the slight pain of
duplication.

**Comment by septract** [comment]
Agree. We should probably colocate `example-archive` with the CN test-cases that are already in the cerberus repo [here](https://github.com/rems-project/cerberus/tree/master/tests/cn).

**Comment by bcpierce00** [comment]
One other potential downside is that keeping examples in two places may
make it slightly more difficult / slightly less automatic for people
changing CN to check for regressions in the tutorial.  But maybe then the
answer is actually to move the tutorial into the CN repo as well...

    - B

**Comment by cp526** [comment]
That's what I was thinking too: having everything, including the tutorial, in one place will make the CI setup smoother.
